crates/aptos/src/account/key_rotation.rs

```diff
@@ -0,0 +1,259 @@
+///
+#[derive(Debug, Parser)]
+pub struct RotateKey {
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+
+    #[clap(flatten)]
+    pub(crate) prompt_options: PromptOptions,
+
+    /// Private key encoded in a type as shown in `encoding`
+    #[clap(long)]
+    pub(crate) new_private_key: String,
+}
+
+#[async_trait]
+impl CliCommand<()> for RotateKey {
+    fn command_name(&self) -> &'static str {
+        "RotateKey"
+    }
+
+    async fn execute(self) -> CliTypedResult<()> {
+        let key = self.new_private_key.as_bytes().to_vec();
+        let new_private_key = self
+            .txn_options
+            .encoding_options
+            .encoding
+            .decode_key::<Ed25519PrivateKey>("--new-private-key", key)?;
+
+        let sender_address = self.txn_options.sender_address()?;
+
+        // Get sequence number for account
+        let sequence_number = self.txn_options.sequence_number(sender_address).await?;
+        let auth_key = self.txn_options.auth_key(sender_address).await?;
+
+        let rotation_proof = RotationProofChallenge {
+            account_address: CORE_CODE_ADDRESS,
+            module_name: "account".to_string(),
+            struct_name: "RotationProofChallenge".to_string(),
+            sequence_number,
+            originator: sender_address,
+            current_auth_key: AccountAddress::from_bytes(&auth_key)
+                .map_err(|err| CliError::UnableToParse("auth_key", err.to_string()))?,
+            new_public_key: new_private_key.public_key().to_bytes().to_vec(),
+        };
+
+        let rotation_msg = bcs::to_bytes(&rotation_proof);
+
+        // Signs the struct using both the current private key and the next private key
+        let rotation_proof_signed_by_current_private_key = self
+            .txn_options
+            .private_key()?
+            .sign_arbitrary_message(&rotation_msg.clone().unwrap());
+        let rotation_proof_signed_by_new_private_key =
+            new_private_key.sign_arbitrary_message(&rotation_msg.unwrap());
+
+        let txn_summary = self
+            .txn_options
+            .submit_transaction(aptos_stdlib::account_rotate_authentication_key_ed25519(
+                rotation_proof_signed_by_current_private_key
+                    .to_bytes()
+                    .to_vec(),
+                rotation_proof_signed_by_new_private_key.to_bytes().to_vec(),
+                // Existing public key
+                self.txn_options
+                    .private_key()?
+                    .public_key()
+                    .to_bytes()
+                    .to_vec(),
+                // New public key
+                new_private_key.public_key().to_bytes().to_vec(),
+            ))
+            .await
+            .map(TransactionSummary::from)?;
+
+        let string = serde_json::to_string_pretty(&txn_summary)
+            .map_err(|err| CliError::UnableToParse("trasaction summary", err.to_string()))?;
+
+        eprintln!("{}", string);
```

Comment:
We don't need to print this out here.  If we want to provide the information about the rotation, we should at the end.

Reply:
Just followed the "convention"